tests/test_engine.py

```diff
@@ -150,6 +169,19 @@ def test_nested_transaction(self):
         self.successResultOf(trx1.close())
         assert not conn.in_transaction()
 
+    def test_savepoints(self):
+        conn = self.get_connection()
+        assert not conn.in_transaction()
+        self.successResultOf(conn.execute(
+            "create table effects (which integer)"
+        ))
+        txn = self.successResultOf(conn.begin())
+        self.successResultOf(conn.execute("insert into effects values (1)"))
+        save = self.successResultOf(conn.begin_nested())
+        self.successResultOf(conn.execute("insert into effects values (2)"))
+        self.successResultOf(save.rollback())
+        self.successResultOf(txn.commit())
```

Comment:
It'd be nice to select out the contents of `effects` and assert that it's only `1` (i.e. to make sure the rollback only affected `2`), more as a test of the test than anything else.

Reply:
Oops, that was the intention there. Will add.